### PR TITLE
add a 404 page to the post

### PR DESCRIPTION
**Pull request by aliceriot**
#### What are the relevant tickets?

#215 

#### What's this PR do?

This implements a 'not found' component and makes some changes so we can show it on the post page, when we get a 404 back from Reddit.

#### How should this be manually tested?

Go to a post that doesn't exist, and you should see a nice 404 page instead of just a generic error.

You should be able to do anything else you would normally be able to do with a post

### CODE REVIEW

**Comment by noisecapella** [comment]
If I click one of the channel links in the sidebar I get redirected to an error page:

screenshot from 2018-01-05 16-23-32

File: static/js/containers/PostPage.js [hunk 1]
```diff
@@ -286,9 +296,10 @@ class PostPage extends React.Component<*, void> {
           title="Delete Post"
           submitText="Yes, Delete"
         >
-          Are you sure you want to delete this post?
+            Are you sure you want to delete this post?
         </Dialog>
         <Card>
+          <div>NO</div>
```
Comment (noisecapella) [inline static/js/containers/PostPage.js]: Haha I think this was meant for the dialog

File: static/js/util/rest_test.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
+// @flow
+import { assert } from "chai"
+
+import { anyProcessing, allLoaded, anyError, anyErrorExcept404 } from "./rest"
+
+describe("rest utils", () => {
+  describe("anyProcessing", () => {
+    it("should return false if nothing has processing true", () => {
```
Comment (noisecapella) [inline static/js/util/rest_test.js]: Can you add a test where `processing: false` and another object has `processing:true`?

File: static/scss/not-found.scss [hunk 3]
```diff
@@ -0,0 +1,15 @@
+.not-found {
```
Comment (noisecapella) [inline static/scss/not-found.scss]: Is there a design for this?

Comment (aliceriot) [reply static/scss/not-found.scss]: image_1024_720

**Comment by aliceriot** [comment]
that issue with the channel page is also an issue on master, I was planning to address it when I did #408

File: static/js/util/rest_test.js (hunk 2 above)
Comment (aliceriot) [reply static/js/util/rest_test.js]: sure

File: static/js/containers/PostPage.js (hunk 1 above)
Comment (aliceriot) [reply static/js/containers/PostPage.js]: oh oops haha

File: static/scss/not-found.scss (hunk 3 above)
Comment (aliceriot) [reply static/scss/not-found.scss]: (I don't think this mock is posted on invision yet)

**Comment by aliceriot** [comment]
@noisecapella comments should be addressed!

**Pull request merged**